lib/improver/nbhood.py

```diff
@@ -55,107 +59,243 @@ class BasicNeighbourhoodProcessing(object):
     # Max extent of kernel in grid cells.
     MAX_KERNEL_CELL_RADIUS = 500
 
-    def __init__(self, radius_in_km, unweighted_mode=False):
+    def __init__(self, radii_in_km, lead_times=None, unweighted_mode=False):
         """
         Create a neighbourhood processing plugin that applies a smoothing
         kernel to points in a cube.
 
         Parameters
         ----------
 
-        radius_in_km : float
-            The radius in kilometres of the neighbourhood kernel to
+        radii_in_km : float or List (if defining lead times)
+            The radii in kilometres of the neighbourhood kernel to
             apply. Rounded up to convert into integer number of grid
             points east and north, based on the characteristic spacing
             at the zero indices of the cube projection-x/y coords.
-
+        lead_times : None or List
+            List of lead times or forecast periods, at which the radii
+            within radii_in_km are defined.
         unweighted_mode : boolean
             If True, use a circle with constant weighting.
             If False, use a circle for neighbourhood kernel with
             weighting decreasing with radius.
 
         """
-        self.radius_in_km = float(radius_in_km)
+        if isinstance(radii_in_km, list):
+            self.radii_in_km = map(float, radii_in_km)
+        else:
+            self.radii_in_km = float(radii_in_km)
+        self.lead_times = lead_times
+        if self.lead_times is not None:
+            if len(radii_in_km) != len(lead_times):
+                msg = ("There is a mismatch in the number of radii "
+                       "and the number of lead times. "
+                       "Unable to continue due to mismatch.")
+                raise ValueError(msg)
         self.unweighted_mode = bool(unweighted_mode)
 
     def __str__(self):
-        result = ('<NeighbourhoodProcessing: radius_in_km: {};' +
+        result = ('<NeighbourhoodProcessing: radii_in_km: {};' +
                   'unweighted_mode: {}>')
         return result.format(
-            self.radius_in_km, self.unweighted_mode)
+            self.radii_in_km, self.unweighted_mode)
+
+    @staticmethod
+    def _find_required_lead_times(cube):
+        """
+        Determine the lead times within a cube, either by reading the
+        forecast_period coordinate, or by calculating the difference between
```

Comment:
Maybe have a note to say that it uses forecast_period points rather than forecast period upper bounds (so should be careful with accumulations/means/maxes etc)